rcsbsearchapi/search.py

```diff
@@ -283,9 +265,32 @@ def __init__(self, value: str):
 
         Args:
             value: free-text query
-            negation: find structures without the pattern
         """
-        super().__init__(service=FULL_TEXT_SEARCH_SERVICE, value=value, negation=None)
+        super().__init__(service=FULL_TEXT_SEARCH_SERVICE, params={"value": value})
+
+
+class SequenceQuery(Terminal):
+    """Special case of a terminal for dna, rna, or protein sequence queries"""
+
+    def __init__(self, value: str,
+                 evalue_cutoff: Optional[float] = 0.1,
+                 identity_cutoff: Optional[float] = 0,
+                 sequence_type: Optional[SequenceType] = "protein"
+                 ):
+        """The string value is a target sequence that is searched
+        Args:
+            value: sequence query
+        """
+        if len(value) < MIN_NUM_OF_RESIDUES:
+            raise ValueError("The sequence must contain at least 25 residues")
+        if identity_cutoff > IDENTITY_CUTOFF_MAX_NUMBER:
+            raise ValueError("Identity cutoff should be less than or equal to 1")
```

Comment:
In addition to my earlier comment about this, I'd recommend checking the lower boundary too, i.e.:
```suggestion
        if identity_cutoff < 0.0 or identity_cutoff > 1.0:
            raise ValueError("Identity cutoff should be between 0 and 1 (inclusive)")
```

Reply:
A comment on this validation: it's not a good idea to have validation in the client unless the constraints can be derived from the API schema programmatically. Hardcoded constants can easily get out of sync and this presents maintenance issue.